kroxylicious-filters/kroxylicious-oauthbearer-validation/src/main/java/io/kroxylicious/proxy/filter/oauthbearer/OauthBearerValidationFilter.java

```diff
@@ -62,14 +64,17 @@ public class OauthBearerValidationFilter
         implements SaslHandshakeRequestFilter, SaslAuthenticateRequestFilter,
         SaslAuthenticateResponseFilter {
 
-    private static final Logger LOGGER = LoggerFactory.getLogger(OauthBearerValidationFilter.class);
+    private static final String UNKNOWN_AUTHORIZATION_ID = "<UNKNOWN AUTHORIZATION_ID>";
 
+    private static final Logger LOGGER = LoggerFactory.getLogger(OauthBearerValidationFilter.class);
+    private static final SaslAuthenticationException INVALID_SASL_STATE_EXCEPTION = new SaslAuthenticationException("invalid sasl state");
```

Comment:
I don't really see the need to have this as a constant.

If we really are going to cache exceptions maybe we should `setCause(null)` to avoid retaining a whole stacktrace in memory? It's not like the stack trace is going to meaninfully reflect the call state at the point it's thrown.